test/apps/mongoose/Makefile

```diff
@@ -28,7 +28,17 @@ $(MONGOOSE_O): $(MONGOOSE_FILES)
 	$(CC) -c $(MONGOOSE_C) $(CFLAGS) -o $@
 
 $(MONGOOSE_FILES): | $(MONGOOSE_DIR)
-	wget -O $@ "https://raw.githubusercontent.com/cesanta/mongoose/7.13/$(notdir $@)"
+	@if [ ! -d "/tmp" ]; then \
+		sudo mkdir /tmp && \
+		sudo chmod 1777 /tmp || { \
+			echo "Error: Failed to create /tmp directory."; \
+			exit 1; \
+		}; \
+	fi && \
+	TARGET_FILE=$(notdir $@) && \
+	TEMP_FILE="/tmp/$$TARGET_FILE" && \
+	wget --continue -O "$$TEMP_FILE" "https://raw.githubusercontent.com/cesanta/mongoose/7.13/$$TARGET_FILE" && \
+	mv "$$TEMP_FILE" "$@" 
```

Comment:
Do not repeat yourself. Wrap the logic above as a reusable script named `atomic_wget`. Add it to the `tools/` directory. And use the script in all build and CI code.

Reply:
This is much better. Thank you for your instruction!